### PR TITLE
add play turns

### PR DESCRIPTION
**Pull request by kudukacar**
Players take turns until the board is full.

### CODE REVIEW

File: spec/board_spec.rb [hunk 1]
```diff
@@ -28,4 +28,26 @@
       end
     end
   end
+
+  describe "#next_token" do
```
Comment (damonkelley) [inline spec/board_spec.rb]: Is `next_token` used outside of the `Board` class?

File: lib/board.rb [hunk 2]
```diff
@@ -21,4 +21,8 @@ def next_token
   def is_available?(pos)
     @board[pos - 1].nil?
   end
+
+  def game_over?
+    @board.none? { |pos| pos.nil? }
```
Comment (damonkelley) [inline lib/board.rb]: You might be able to write this as `@board.none?(&:nil?)` as well. No change needed, just wanted to throw that out there.

Comment (damonkelley) [reply lib/board.rb]: I saw on the last PR you indicated that you changed all of the classes except for `Board` use attr_readers/attr_writers. Was there something about the `Board` that made this difficult here?

File: lib/tictactoe.rb [hunk 3]
```diff
@@ -15,10 +15,10 @@ def initialize(presenter, display, board, player)
...
   def run
     show_board
-    play_turn
-    show_board
-    play_turn
-    show_board
+    until board.game_over?
+      play_turn
+      show_board
+    end
```
Comment (damonkelley) [inline lib/tictactoe.rb:21]: Cool 😎

File: lib/board.rb (hunk 2 above)
Comment (kudukacar) [reply lib/board.rb]: I didn't want to put an attr_accessor and have the ability to modify board.  I guess I could put a private attr_accessor (somehow was getting some errors), but figured just same as having an instance variable.  I'll give a shot again.

File: spec/board_spec.rb (hunk 1 above)
Comment (kudukacar) [reply spec/board_spec.rb]: It isn't.  I initially took out the spec, but then I put it back in.  In the future I might need to use next_token from outside Board, and wanted to indicate that it's a stable method.

File: lib/board.rb (hunk 2 above)
Comment (kudukacar) [reply lib/board.rb]: I guess you only need an attr_reader, and it works.  I made the change.

File: spec/board_spec.rb (hunk 1 above)
Comment (kudukacar) [reply spec/board_spec.rb]: Do you think I should take out the spec?

File: lib/board.rb (hunk 2 above)
Comment (damonkelley) [reply lib/board.rb]: 👍 

In situations when you want a public reader and a private writer you can marke the  `attr_writer` as private.

File: spec/board_spec.rb (hunk 1 above)
Comment (damonkelley) [reply spec/board_spec.rb]: Yeah, I would recommend taking it out of the spec and explicitly making it private. That will make it easier for us to change if we need to. If/when we need to make it public we can.

Comment (kudukacar) [reply spec/board_spec.rb]: Sounds good.  Made the change.

**Comment by damonkelley** [comment]
🚢

**Pull request merged**